webui/src/components/RepositoryCreateForm.js

```diff
@@ -18,20 +18,33 @@ export const RepositoryCreateForm = connect(
     };
   })(({ error, onSubmit, onCancel, create, sm = 6, config }) => {
     const fieldNameOffset = 3;
+    const repoValidityRegex = /^[a-z0-9][a-z0-9-]{2,62}$/;
+    const storageNamespaceValidityRegex = /^(s3|gs|mem|local|transient):\/.*$/;
 
     const [formValid, setFormValid] = useState(false);
+    const [repoValid, setRepoValid] = useState(true);
+    const [storageNamespaceValid, setStorageNamespaceValid] = useState(true);
+    const [defaultBranchValid, setDefaultBranchValid] = useState(true);
     const storageNamespaceField = useRef(null);
     const defaultBranchField = useRef(null);
     const repoIdField = useRef(null);
 
-    const checkValidity = () => {
-        if (repoIdField.current.value.length === 0 ||
-            storageNamespaceField.current.value.length === 0 ||
-            defaultBranchField.current.value.length === 0) {
-            setFormValid(false);
-            return;
-        }
-        setFormValid(true);
+    const checkRepoValidity = () => {
+        const isRepoValid = repoValidityRegex.test(repoIdField.current.value)
+        setRepoValid(isRepoValid);
+        setFormValid(isRepoValid);
```

Comment:
Here and elsewhere.
This is wrong, form validity should be AND of all three, not the state of the last validity function running.